pkg/reconciler/workload/resource/resource_reconcile.go

```diff
@@ -93,17 +117,16 @@ func (c *Controller) reconcileResource(ctx context.Context, lclusterName logical
 
 		syncTargetKey := strings.TrimPrefix(f, syncershared.SyncerFinalizerNamePrefix)
 		logger = logger.WithValues("syncTargetKey", syncTargetKey)
-		objs, err := c.syncTargetIndexer.ByIndex(indexers.SyncTargetsBySyncTargetKey, syncTargetKey)
+		syncTarget, err := c.getSyncTargetFromKey(syncTargetKey)
```

Comment:
I can't comment on L112 but please change it to `logger := logger.WitValues("finalizer", f)`

Reply:
done